tls/s2n_kem.h

```diff
@@ -76,6 +82,10 @@ struct s2n_kem_group {
     uint16_t iana_id;
     const struct s2n_ecc_named_curve *curve;
     const struct s2n_kem *kem;
+
+    /* Whether the PQ KeyShare should be sent before the ECC KeyShare. Only enabled for X25519MLKEM768.
+     * See: https://datatracker.ietf.org/doc/html/draft-kwiatkowski-tls-ecdhe-mlkem-02#name-negotiated-groups */
```

Comment:
This still seems odd to me :/ Like I understand the reasoning, but it seems like one of those workarounds we're going to regret a decade from now. But I guess can't argue with the RFC. Any chance this is going to change before mlkem makes it out of draft?